operator/operator.go

```diff
@@ -160,6 +161,75 @@ func (operator *Operator) handleBlocks() error {
 	return nil
 }
 
+func (operator *Operator) handleChallenges() error {
+	challenges := make(chan *bindings.ErdstallChallenged)
+	sub, err := operator.contract.WatchChallenged(nil, challenges, nil, nil)
+	if err != nil {
+		return fmt.Errorf("creating challenge subcription: %w", err)
+	}
+	defer sub.Unsubscribe()
+
+	for {
+		select {
+		case err := <-sub.Err():
+			return fmt.Errorf("subscription error: %w", err)
+
+		case c := <-challenges:
+			log.Warnf("Operator.handleChallenges: Incoming challenge for account %v and epoch %v", c.Account.String(), c.Epoch)
+
+			tr, err := func() (*bind.TransactOpts, error) {
+				ctx, cancel := createDefaultContext()
+				defer cancel()
+
+				return operator.ethClient.NewTransactor(ctx, big.NewInt(0), eth.DefaultGasLimit, operator.ethClient.Account())
+			}()
+			if err != nil {
+				log.Errorf("Operator.handleChallenges: Failed to create transactor for response to challenge %v: %v", c, err)
+				continue
+			}
+
+			balanceProof, ok := operator.balanceProofs.Get(c.Account)
+			if !ok {
+				log.Error("Operator.handleChallenges: Failed to get balance proof for challenge response")
+				continue
+			}
+
+			balance := bindings.ErdstallBalance{
+				Epoch:   balanceProof.Balance.Epoch,
+				Account: balanceProof.Balance.Account,
+				Value:   balanceProof.Balance.Value,
+			}
+
+			tx, err := operator.contract.Exit(tr, balance, balanceProof.Sig)
+			if err != nil {
+				log.Errorf("Operator.handleChallenges: Failed to send response to challenge %v: %v", c, err)
+				continue
+			}
+
+			// Track challenge response transaction status
+			//TODO: do not create a new go routine for every transaction, run a dedicated go routine that tracks all?
```

Comment:
I think it's fine how you do it for now. There shouldn't be that many challenges happening at the same time that this becomes an issue at this stage.

Reply:
Agreed, we don't have to take care of it now. However, I would like to keep note of that so that we do not forget about it later. I would therefore keep the comment, or, alternatively, creating an issue for hat.

Reply:
Created #50

Reply:
I prefer creating an issue so we don't have rotting TODO comments.